### DISCUSSION TITLE
Sunbird RC Community Call to discuss tickets 317, 318, 319, 320

### DISCUSSION THREAD

**Discussion by sukhpreetssekhon** [Sunbird-RC/community#322] [General]
Hi Everyone - We plan to have a Sunbird RC community call on Monday 16 January from 2:00-3:30pm to discuss the following tickets. If you are interested in these issues or have inputs on the architecture or use case relating to these please join the following zoom link for the call - https://us06web.zoom.us/j/83300902256?pwd=OWZWZXJyUFJFMWhUYXIwbXF5dXFZQT09

- identity management
https://github.com/orgs/Sunbird-RC/discussions/320

- programmatic credential issuance and credential issuance by individuals
https://github.com/orgs/Sunbird-RC/discussions/319

- credential transfer between wallets, pulling in credentials from different sources
https://github.com/orgs/Sunbird-RC/discussions/318

- security of the issuer private key
https://github.com/orgs/Sunbird-RC/discussions/317

**Comment by sukhpreetssekhon** [comment]
@pramodkvarma @surendrasinghs please could you share the link of the video recording of this call. Thanks

**Reply by surendrasinghs** [reply]
Please access it here https://us06web.zoom.us/rec/share/9CBudyNSfUbuUjFOieL71NPYNSUcCpuS8WyA5TLI1sjXW1bJkM8u5rtd-7OTEddc.med_ta5d44zkI9Nw 
Passcode: ^B22L$uU

**Comment by sukhpreetssekhon** [comment]
Thank you everyone for the discussion earlier this week. Two key themes emerged.

**1. Having a clean VC API and better decoupling from keycloak**
1.1 Credential store
1.2 Verifiable credential as identity
1.3 Verification of credentials (without auth)
1.4. Wallet for a user (cloud based)
     
**2. Having a PKI layer to be holder identity aware**
2.1 DID as a collection of verifiable identity credentials
2.2 DID management via auth into any one identity providers
2.3 Consent based sharing of credential(s)
2.4 Verification of credentials (with auth)
-- by third party with consent
-- for self retrieval with auth using ID mentioned on the credential_
2.5. Immutability
2.6. Wallet for a user (device based)

For ULP, we need 1 in time and we will post RFCs for the same on the RC community shortly, 2 can be a continued effort decoupled from ULP timelines to evolve RC to become more identity aware. 

cc - @rahul101001000 @ChakshuGautam @parthlawate @coolbung @chakshu @smohan-dw @girirajdaga

**Reply by csamprajan** [reply]
In reference to the second aspect of the holder identity aware PKI layer, I would like to suggest a solution that fulfills all the necessary requirements. I kindly request your valuable time to review my proposal.

Decentralized ID solutions are gaining acceptance in several countries, with Europe recently releasing specifications for the eIDAS 2.0 identity credential for its citizens. 

Taking Aadhar as a baseline, a unique DID (Super Aadhar) could be issued, and the proposed solution includes the following components:

- Creation of a unique, resolvable, W3C-compliant DID for every citizen, using the did:web method.
- The citizen will have control over their Private Key and Recovery Keys, associated with their DID.
- The citizen's Public Key will be shared with the government, which will then host it on a dedicated government website.


 
![Aadhar VC issuance](https://user-images.githubusercontent.com/47203553/230247637-82e415b9-c7dd-4a6d-b5d2-470264622762.svg)

An aadhar based citizen’s DID could look something like​
`did:web:< Aadhar’s website >:aadhar:< unique Id >​`

All other VCs related to that citizen could be linked to this DID​​. For eg.​​

Driver’s license:​​
Issuer – `did:web:< transportation ministry’s website >:< state >​`
Holder – `did:web:< aadhar’s website >:aadhar:< unique Id >​​`
​
Work experience letter from a private company​
Issuer – `did:web:<company’s website>:india​​`
Holder – `did:web:< aadhar’s website >:aadhar:< unique Id >`


**One DID - Multiple Keys:**

A DID document can contain multiple keys, each with its own unique identifier.

<img width="1118" alt="Screenshot 2023-04-06 at 6 41 32 AM" src="https://user-images.githubusercontent.com/47203553/230247853-956f8501-59f5-4306-90e8-b572ad80d648.png">


A Verifiable Credential (VC) can reference the unique identifier of the key with which it was signed, by using the 'proof.verificationMethod' attribute.
https://www.w3.org/TR/vc-data-model/#:~:text=13T18%3A19%3A39Z%22%2C%0A%20%20%20%20%22-,verificationMethod,-%22%3A%20%22https%3A//example.edu

**Key Revocation / Rotation **

Q: What happens if a user loses their phone and cannot access their Edge Wallet, resulting in the loss of their private key?

A: The user can install a different wallet on a new phone and generate a new key pair (private key and public key). The new public key will then be shared with the government, who will add it to the DID document and publish the changes to the decentralized ledger or hosted URL. The citizen can request that the government remove any old public keys from DID document. Any new VC will be signed using new key private key.



**did methods comparision (why choose did:web):**

<a class="CID5199699a-c8c8-47ff-80f7-cd155a88a25a" data-sessionid="9b5037bb-e089-9fbd-3ed8-877dd20e90b4" data-shapeids="3" data-slideid="" style="color: rgb(0, 0, 0); font-family: Times; font-size: medium; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"> </a><div style="color: rgb(0, 0, 0); font-family: Times; font-size: medium; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">

​ | Public Block Chain​ | Federated Block chain​ | DID:WEB​
-- | -- | -- | --
Ease of implementation ​ | Not easy​ | ​Difficult​​ | Very easy​​
Cost​ | Expensive​ | Moderate​ | Negligible​​
Scalability​ | Best of the block chains cannot do more than 1000 txns per second​ | Same problems as public chain​ | Scales very well​
Security / Reliability​ | Battle tested. No public chain has been hacked yet.​ | Not as safe as public chain​ | Website on which public key is hosted is the single point of failure​

</div>



**FAQs:**

Q: Why is it necessary to link all Verifiable Credentials (VCs) to a single holder DID? Can't there be multiple context-aware identities?

A: Using multiple identities for a single holder may result in redundant claims in VCs. For instance, a person's name may be listed differently on their driver's license versus their work experience letter. This presents a challenge in establishing that the two names refer to the same person. Moreover, if a user changes their name, all of their VCs must be updated. Having a single holder DID for all VCs ensures that a single private key could be used to combine several VCs into a verifiable presentation (VP).

Q: Why is it important for the holder DID to be linked to Aadhar?

A: If the holder DID is issued by a private company, establishing trust becomes difficult. However, a DID issued by the government is trusted internationally. Aadhar provides a suitable solution for this purpose.

Q: How can a citizen store Verifiable Credentials (VCs) for their family members on their wallet?

A: The wallet can be designed to allow citizens to store VCs that are not necessarily signed by them. However, in order to act on behalf of a family member, the W3C specifications allow for 'capability delegation'.
https://www.w3.org/TR/vc-data-integrity/#:~:text=popular%20encryption%20libraries.-,capabilityDelegation,-Indicates%20that%20the
Example implementation: https://w3c-ccg.github.io/did-method-key/#:~:text=z6MkhaXgBZDvotDkL5257faiztiGiC2QtKLGpbnnEGta2doK%22%0A%20%20%5D%2C%0A%20%20%22-,capabilityDelegation,-%22%3A%20%5B%0A%20%20%20%20%22did

@pramodkvarma @rahul101001000 @smohan-dw @surendrasinghs  @girirajdaga
